src/core/core.interaction.js

```diff
@@ -42,38 +42,54 @@ function evaluateAllVisibleItems(chart, handler) {
 }
 
 /**
- * Helper function to check the items at the hovered index on the index scale
+ * Helper function to do binary search when possible
+ * @param {object} metaset - the dataset meta
+ * @param {string} axis - the axis mide. x|y|xy
+ * @param {number} value - the value to find
+ * @param {boolean} intersect - should the element intersect
+ * @returns {lo, hi} indices to search data array between
+ */
+function binarySearch(metaset, axis, value, intersect) {
+	const {controller, data, _sorted} = metaset;
+	const iScale = controller._cachedMeta.iScale;
+	if (iScale && axis === iScale.axis && _sorted) {
+		const lookupMethod = iScale._reversePixels ? _rlookup : _lookup;
+		if (!intersect) {
+			return lookupMethod(data, axis, value);
+		} else if (controller._sharedOptions) {
```

Comment:
Why do we need `_sharedOptions` defined? It doesn't look like it's used. Should we check instead if `data[0].getRange` is defined?

The other thing I'm wondering about is that it looks like it only gets initialized on a few chart types

Reply:
if the config has something like `radius: function(ctx) { return ctx.index * 10; }`, then checking first points range does not help much. So `_sharedOptions` tells us each point is has equal options and the range would be same for all.

Only few chart types have benefit from `binarySearch`. It starts to be noticeable at around 1000 elements.

Reply:
Ah, that makes sense. Would you mind adding a comment explaining why `_sharedOptions` is needed? I missed it the first time, so am afraid others might as well

Reply:
Done